tests/_internal/pydantic/test_type_adapter.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+import pytest
+from pydantic import BaseModel
+
+from prefect._internal.pydantic import validate_python
+from prefect._internal.pydantic._flags import EXPECT_DEPRECATION_WARNINGS
+
+
+class Model(BaseModel):
+    a: int
+    b: str
+    c: bool
+
+
+@pytest.mark.skipif(
+    EXPECT_DEPRECATION_WARNINGS,
+    reason="Valid when compatibility is enabled or v1 is installed",
+)
+@pytest.mark.parametrize(
+    "type_, value, expected",
+    [
+        (int, 42, 42),
+        (str, "42", "42"),
+        (bool, True, True),
+        (Model, {"a": 42, "b": "42", "c": True}, Model(a=42, b="42", c=True)),
+        (List[Model], [{"a": 42, "b": "42", "c": True}], [Model(a=42, b="42", c=True)]),
+    ],
+)
+def test_validate_python(type_, value, expected):
+    assert validate_python(type_, value) == expected
+
+
+@pytest.mark.skipif(
+    not EXPECT_DEPRECATION_WARNINGS,
+    reason="Valid when compatibility is disabled and v2 is installed",
+)
+@pytest.mark.parametrize(
+    "type_, value, expected",
+    [
+        (int, 42, 42),
+        (str, "42", "42"),
+        (bool, True, True),
+        (Model, {"a": 42, "b": "42", "c": True}, Model(a=42, b="42", c=True)),
+        (List[Model], [{"a": 42, "b": "42", "c": True}], [Model(a=42, b="42", c=True)]),
+    ],
+)
+def test_validate_python_with_deprecation_warnings(type_, value, expected):
```

Comment:
Is it possible to just add a docstring here for ppl like me who have trouble with "skip if not true" logic? 

Something like "deprecation warnings occur when v2 installed but not used here" would be super helpful for me. Kinda same deal for above

Reply:
sure!! when i was trying to do the skipif logic using HAS_PYDANTIC_2 and USE_PYDANTIC_2 i was going crazy for that same reason bc skip is a negative haha

will add docstrings! good callout